### DISCUSSION TITLE
Is it possible to add server from agents using private ip?

### DISCUSSION THREAD

**Discussion by heanzyzabala** [k3s-io/k3s#4721] [General]
Currently https is required when adding `K3S_URL`. Running on low budget here, can't really afford to buy a load balancer. Nginx + Let's encrypt is also an option.

**Comment by brandond** [comment]
You don't need a certificate issued by a public CA for the server; it is actually expected that the registration endpoint used for K3S_URL will just be one of the servers, using a cert signed by the cluster root CA. Similarly, you don't have to provide a load-balancer in front of the servers - you can absolutely point agents at one of the servers IPs, as long as you are OK with that server being a single point of failure for joining agents.

**Reply by heanzyzabala** [reply]
@brandond It worked. Thanks!
